precompiles/referenda/src/lib.rs

```diff
@@ -379,11 +420,37 @@ where
 	/// * index: The index of a closed referendum whose Submission Deposit has not yet been refunded.
 	#[precompile::public("refundSubmissionDeposit(uint32)")]
 	fn refund_submission_deposit(handle: &mut impl PrecompileHandle, index: u32) -> EvmResult {
+		handle.record_log_costs_manual(1, 32 * 3)?;
+		// Get refunding deposit before dispatch
+		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+		let refunded_deposit: U256 = match ReferendumInfoFor::<Runtime>::get(index)
+			.ok_or(RevertReason::custom("Referendum index does not exist").in_field("index"))?
+		{
+			ReferendumInfo::Approved(_, Some(s), _) | ReferendumInfo::Cancelled(_, Some(s), _) => {
+				s.amount.into()
+			}
+			// We let the pallet handle the RenferendumInfo validation logic on dispatch.
+			_ => U256::zero(),
+		};
+
 		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
 
 		let call = ReferendaCall::<Runtime>::refund_submission_deposit { index }.into();
 
 		<RuntimeHelper<Runtime>>::try_dispatch(handle, Some(origin).into(), call)?;
+
+		let event = log1(
+			handle.context().address,
+			SELECTOR_LOG_SUBMISSION_DEPOSIT_REFUNDED,
+			EvmDataWriter::new()
+				.write::<u32>(index)
+				.write::<Address>(Address(handle.context().caller))
```

Comment:
```suggestion
				.write::<Address>(Address(handle.context().caller))
```
Should be recipient of refunded deposit, not caller